nautobot_netbox_importer/diffsync/models/locations.py

```diff
@@ -3,119 +3,111 @@
-        setattr(target, field.nautobot.name, result)
-
-    field.set_importer(importer)
+    field.set_nautobot_field(field.name)
+    field.set_importer(location_importer)
+    field.handle_sibling("site", field.nautobot.name)
+    field.handle_sibling("region", field.nautobot.name)
 
 
 def _define_location_parent(field: SourceField) -> None:
-    field.set_nautobot_field(field.name)
-    field.handle_sibling("site")
-
-    site_wrapper = field.wrapper.adapter.wrappers["dcim.site"]
+    wrapper = field.wrapper
+    site_wrapper = wrapper.adapter.wrappers["dcim.site"]
 
-    def importer(source: RecordData, target: DiffSyncBaseModel) -> None:
+    def location_parent_importer(source: RecordData, target: DiffSyncBaseModel) -> None:
         parent = source.get(field.name, None)
         site = source.get("site", None)
 
         if parent:
-            result = field.wrapper.get_pk_from_uid(parent)
-            field.wrapper.add_reference(result, field.wrapper)
+            result = wrapper.get_pk_from_uid(parent)
+            wrapper.add_reference(wrapper, result)
         elif site:
             result = site_wrapper.get_pk_from_uid(site)
-            site_wrapper.add_reference(result, field.wrapper)
+            wrapper.add_reference(site_wrapper, result)
         else:
             return
 
         setattr(target, field.nautobot.name, result)
 
-    field.set_importer(importer)
+    field.set_nautobot_field(field.name)
+    field.set_importer(location_parent_importer)
+    field.handle_sibling("site", field.nautobot.name)
 
 
 def _define_site_group(field: SourceField) -> None:
-    site_group_wrapper = field.wrapper.adapter.wrappers["dcim.sitegroup"]
-    location_type_wrapper = field.wrapper.adapter.wrappers["dcim.locationtype"]
+    wrapper = field.wrapper
+    site_group_wrapper = wrapper.adapter.wrappers["dcim.sitegroup"]
+    location_type_wrapper = wrapper.adapter.wrappers["dcim.locationtype"]
     site_type_uid = location_type_wrapper.get_pk_from_uid("Site")
 
     field.set_nautobot_field("location_type")
 
-    def importer(source: RecordData, target: DiffSyncBaseModel) -> None:
+    def site_group_importer(source: RecordData, target: DiffSyncBaseModel) -> None:
         group = source.get(field.name, None)
         if group:
             result = site_group_wrapper.get_pk_from_uid(group)
-            site_group_wrapper.add_reference(result, field.wrapper)
+            wrapper.add_reference(site_group_wrapper, result)
         else:
             result = site_type_uid
-            location_type_wrapper.add_reference(result, field.wrapper)
+            wrapper.add_reference(location_type_wrapper, result)
 
         setattr(target, field.nautobot.name, result)
 
-    field.set_importer(importer)
+    field.set_importer(site_group_importer)
 
 
 def setup(adapter: SourceAdapter) -> None:
     """Setup locations for NetBox importer."""
+
+    def forward_references(wrapper: SourceModelWrapper, references: SourceReferences) -> None:
```

Comment:
a description of what is doing could help to understand it

Reply:
Added a description.